app/decorators/models/discussion_topic/materialized_view_decorator.rb

```diff
@@ -0,0 +1,30 @@
+DiscussionTopic::MaterializedView.class_eval do
+  def strongmind_materialized_view_json(opts = {})
+    graded_discussion = self.discussion_topic.graded_discussion?
+    opts[:show_other_users_submissions] = false
+    entry_ids = nil
+    if graded_discussion && launch_darkly_render_discussion_entries(opts[:launch_darkly_user])
+      student_submission_graded = self.discussion_topic.assignment.submissions.graded.where(user_id: opts[:user_id]).any?
+      if student_submission_graded
+        opts[:show_other_users_submissions] = true
+        entry_ids = self.entry_ids_array
+      else
+        entries = self.discussion_topic.discussion_entries.where(user_id: opts[:user_id])
+        entry_ids = entries.pluck(:id)
+        ungraded_json_structure = discussion_entry_api_json(entries, discussion_topic.context, nil, nil, [])
+      end
+    else
+      entry_ids = self.entry_ids_array
+    end
+
+    opts[:entry_ids] = entry_ids
+    opts[:json_structure] = ungraded_json_structure.present? ? ungraded_json_structure.to_json : self.json_structure
+    instructure_materialized_view_json(opts)
+  end
+  alias_method :instructure_materialized_view_json, :materialized_view_json
+  alias_method :materialized_view_json, :strongmind_materialized_view_json
+
+  def launch_darkly_render_discussion_entries(ld_user)
+    Rails.configuration.launch_darkly_client.variation("expose-discussion-entries-only-if-graded-submission-exists", ld_user, false)
```

Comment:
should ld_user be using @launch_darkly_user instead?

Reply:
It gets passed in on line 6 through the options, which come from @launch_darkly_user in line 111 in dicussion_topics_api_controller (in the canvas-lms repo). It had to be done that way due to limited scoping of this view